plugins/Actions/Archiver.php

```diff
@@ -468,35 +470,71 @@ protected function insertPageUrlsReports()
      */
     protected function archiveDayActionsGoals(int $rankingQueryLimit): void
     {
-        $this->archiveDayActionsGoalsPages($rankingQueryLimit,true);
-        $this->archiveDayActionsGoalsPages($rankingQueryLimit,false);
+
+        $site = $this->getProcessor()->getParams()->getSite();
+
+        if (!\Piwik\Common::isGoalPluginEnabled() ||
+            GeneralConfig::getConfigValue('disable_archive_actions_goals', $site->getId())) {
+            return;
+        }
+
+        $goals = $this->getGoalsForSite($site->getId());
+
+        if ($site->isEcommerceEnabled()) {
+            $goals[] = 0;
```

Comment:
can we use the constant for the `0` goalId so it's clear what it's about? And do we also need to add the constant for abandoned cards which is `-1`?

Reply:
I've updated this to use the constants from GoalManager and also included the abandoned carts metric automatically if the site is setup for ecommerce.